tls/s2n_server_hello.c

```diff
@@ -128,6 +128,13 @@ static int s2n_server_hello_parse(struct s2n_connection *conn)
         S2N_ERROR_IF(session_id_len != conn->session_id_len || memcmp(session_id, conn->session_id, session_id_len), S2N_ERR_BAD_MESSAGE);
         conn->actual_protocol_version = conn->server_protocol_version;
         GUARD(s2n_set_cipher_as_client(conn, cipher_suite_wire));
+        /* From RFC section: https://tools.ietf.org/html/rfc8446#section-4.2.11:
+         * Client MUST verify that the server selected a cipher suite indicating a Hash associated with the PSK 
+         * */
+        if (conn->psk_params.chosen_psk) {
+            ENSURE_POSIX(conn->secure.cipher_suite->prf_alg == conn->psk_params.chosen_psk->hmac_alg,
+                         S2N_ERR_HASH_INVALID_ALGORITHM);
+        }
```

Comment:
Yes, I did consider moving this check into `s2n_set_cipher_as_client`, it would definitely make it easier for testing. This could be the last check within `s2n_set_cipher_as_client` prior to setting the cipher suite in `conn->secure.cipher_suite`. Ideally the check is better suited within s2n_server_psk_recv, but. the extensions are executed prior to setting `conn->secure.cipher_suite` value.